protocols/session/src/protocol/mod.rs

```diff
@@ -90,7 +90,7 @@ impl<const C: usize> SessionMessage<C> {
     /// and two bytes for the message length.
     pub const HEADER_SIZE: usize = 1 + size_of::<SessionMessageDiscriminants>() + size_of::<u16>();
     /// Maximum size of the message in v1.
-    pub const MAX_MESSAGE_LENGTH: usize = 2047;
+    pub const MAX_MESSAGE_LENGTH: usize = C.saturating_sub(Self::HEADER_SIZE);
```

Comment:
Maybe let's make a test to assert that this value is more than a certain amount of bytes?